openpnm/io/JSONGraphFormat.py

```diff
@@ -0,0 +1,185 @@
+import json
+import os
+import pickle
+from pathlib import Path
+
+import jsonschema
+import scipy as sp
+
+from openpnm.utils import logging
+from openpnm.io import GenericIO
+from openpnm.models.geometry import (pore_area, pore_volume, throat_area,
+                                     throat_perimeter, throat_surface_area,
+                                     throat_volume)
+from openpnm.network import GenericNetwork
+
+logger = logging.getLogger(__name__)
+
+
+class JSONGraphFormat(GenericIO):
+    r"""
+    Class for reading and writing OpenPNM networks to JSON Graph Format (JGF).
+
+    Notes
+    -----
+    The JGF standard must contain data formatted according to
+    http://jsongraphformat.info and enforced by JSON schema validation.
+    """
+
+    @classmethod
+    def __validate_json__(self, json_file):
+        # Validate name of schema file
+        relative_path = '../../utils/jgf_schema.pkl'
+        schema_file = Path(os.path.realpath(__file__), relative_path)
+        schema_file = self._parse_filename(filename=schema_file, ext='pkl')
+
+        # Load schema from pickle file
+        with open(schema_file, 'rb') as file:
+            jgf_schema = pickle.load(file)
+
+        # Validate JSON agains schema
+        try:
+            jsonschema.validate(json_file, jgf_schema)
+            return True
+        except jsonschema.exceptions.ValidationError:
+            return False
+
+    @classmethod
+    def save(self, network, filename=''):
```

Comment:
Typically we allow users to optionally send in a list of phases here, which will also save the phase data to the file.  This is optional since maybe they only want the network, or maybe they only want some of the phase, not all.  If you look at one of the other io.save methods you'll see that we use the `Dict` class to actually build all the data into a single `dict` that we then parse into another custom format.   You're pulling data from the network object itself (e.g. line 94-95 below),  so this would be changed to `'x': int(dict_['net.throat.conns'][ts, 0]`.  This would then let you also do `'T': float(dict_['air.pore.temperature'])`.

Reply:
The JGF standard is, in principle, targeted at the network geometry/topology only. One could, if needed, create a JGF-derived standard that includes the phases as `node` or `edge` metadata. But in the best spirit of object orientation, I would rather do this in a separate step by creating a new class derived from `JSONGraphFormat` instead of `GenericIO` to avoid code duplication.

In the meantime, I will add the docstring note you suggested.